cluster-autoscaler/core/scaledown/planner/planner_test.go

```diff
@@ -0,0 +1,401 @@
+				nodeUndergoingDeletion("n4", 2000, 10),
+			},
+			pods: []*apiv1.Pod{
+				scheduledPod("p1", 500, 1, "n2"),
+				scheduledPod("p2", 500, 1, "n2"),
+				scheduledPod("p3", 500, 1, "n2"),
+				scheduledPod("p4", 500, 1, "n4"),
+				scheduledPod("p5", 500, 1, "n4"),
+			},
+			eligible: []string{"n1", "n3"},
+			actuationStatus: &fakeActuationStatus{
+				currentlyDrained: []string{"n2", "n4"},
+			},
+			wantUnneeded: []string{},
+			wantErr:      true,
+		},
+		{
+			name: "multiple drained nodes but new candidates found",
+			nodes: []*apiv1.Node{
+				BuildTestNode("n1", 2000, 10),
+				nodeUndergoingDeletion("n2", 2000, 10),
+				BuildTestNode("n3", 2000, 10),
+				nodeUndergoingDeletion("n4", 2000, 10),
+				BuildTestNode("n5", 2000, 10),
+			},
+			pods: []*apiv1.Pod{
+				scheduledPod("p1", 400, 1, "n1"),
+				scheduledPod("p2", 400, 1, "n2"),
+				scheduledPod("p3", 400, 1, "n3"),
+				scheduledPod("p4", 400, 1, "n4"),
+				scheduledPod("p5", 400, 1, "n5"),
+			},
+			eligible: []string{"n1", "n3", "n5"},
+			actuationStatus: &fakeActuationStatus{
+				currentlyDrained: []string{"n2", "n4"},
+			},
+			wantUnneeded: []string{"n1", "n3"},
+		},
+		{
+			name: "recently evicted pods can schedule elsewhere, node uneeded",
+			nodes: []*apiv1.Node{
+				BuildTestNode("n1", 1000, 10),
+				BuildTestNode("n2", 1000, 10),
+				BuildTestNode("n3", 1000, 10),
+			},
+			pods: []*apiv1.Pod{
+				scheduledPod("p1", 500, 1, "n2"),
+				scheduledPod("p2", 500, 1, "n2"),
+			},
+			eligible: []string{"n1", "n2"},
+			actuationStatus: &fakeActuationStatus{
+				recentEvictions: []*apiv1.Pod{
+					scheduledPod("p3", 500, 1, "n4"),
+				},
+			},
+			wantUnneeded: []string{"n1"},
+		},
+		{
+			name: "recently evicted pods can schedule elsewhere, no unneeded",
+			nodes: []*apiv1.Node{
+				BuildTestNode("n1", 1000, 10),
+				BuildTestNode("n2", 1000, 10),
+				BuildTestNode("n3", 1000, 10),
+			},
+			pods: []*apiv1.Pod{
+				scheduledPod("p1", 500, 1, "n2"),
+				scheduledPod("p2", 500, 1, "n2"),
+			},
+			eligible: []string{"n1", "n2"},
+			actuationStatus: &fakeActuationStatus{
+				recentEvictions: []*apiv1.Pod{
+					scheduledPod("p3", 500, 1, "n4"),
+					scheduledPod("p4", 500, 1, "n4"),
+					scheduledPod("p5", 500, 1, "n4"),
+				},
+			},
+			wantUnneeded: []string{},
+		},
+		{
+			name: "recently evicted pods can't schedule elsewhere",
+			nodes: []*apiv1.Node{
+				BuildTestNode("n1", 1000, 10),
+				BuildTestNode("n2", 1000, 10),
+			},
+			pods: []*apiv1.Pod{
+				scheduledPod("p1", 500, 1, "n1"),
+				scheduledPod("p2", 500, 1, "n1"),
+			},
+			eligible: []string{"n1", "n2"},
+			actuationStatus: &fakeActuationStatus{
+				recentEvictions: []*apiv1.Pod{
+					scheduledPod("p3", 500, 1, "n3"),
+					scheduledPod("p4", 500, 1, "n3"),
+					scheduledPod("p5", 500, 1, "n3"),
+				},
+			},
+			wantUnneeded: []string{},
+			wantErr:      true,
+		},
+		{
+			name: "multiple drained nodes and recent evictions, no unneeded",
```

Comment:
In this case, can't it happen that:
* `p2` + `p6` get assigned to `n1` (200 + 200 + 600 = 1000)
* `p4` + `p7` get assigned to `n3` (200 + 200 + 600 = 1000)

Which would leave `n5` with 20% utilization, thus unneeded? Or am I missing something here?

Reply:
20% utilization is not sufficient for the node to become unneeded: it is not possible to move its pod anywhere else in the cluster, so it has to stay.

Reply:
Ugh, of course. Knew I was missing something :P